### DISCUSSION TITLE
Problem filename after Github Import

### DISCUSSION THREAD

**Discussion by remids** [community/community#22128] [New to GitHub]
<p>I imported a repository using Github’s import, and it worked great, except for one thing.</p>
<p>There was a file in the repo with a special character - the registered sign (‘r’ inside a ciricle).  I’m not sure if the specific character in use is unicode or from a particular Windows character set, but it’s showing up in Github for Windows as the replacement character (a question mark inside a black diamond).  Trying to use Git on the command line, it shows as ‘\256’.</p>
<p>I renamed the file to remove the problem character, and pushed the commit.</p>
<p>When I clone the repo, and I’m now seeing:</p>
<p>warning: the following paths have collided (e.g. case-sensitive paths<br>
on a case-insensitive filesystem) and only one from the same<br>
colliding group is in the working tree:</p>
<p>‘Resources/DC/DC® SFTP Download Utility.pdf’</p>
<p>If I try to view this file on <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a>, I get HTTP ERROR 400.</p>
<p>I can’t remove it in Github for Windows - it keeps showing as Deleted, but when I commit it always sticks around afterwards still showing as deleted.  I can’t add it to the .gitignore to ignore it, nor can I discard changes on it.</p>
<p>On the command line, here are things I’ve tried and the output (note, the filespec used in the second git rm attempt is the filename that’s displayed in Github for Windows):</p>
<blockquote>
<p>&gt; git status<br>
On branch master<br>
Your branch is up to date with ‘origin/master’.</p>
<p>Changes not staged for commit:<br>
(use “git add/rm …” to update what will be committed)<br>
(use “git restore …” to discard changes in working directory)<br>
deleted:    “Resources/DC/DC\256 SFTP Download Utility.pdf”</p>
<p>no changes added to commit (use “git add” and/or “git commit -a”)</p>
<p>&gt;git rm “Resources/DC/DC\256 SFTP Download Utility.pdf”<br>
fatal: pathspec ‘Resources/DC/DC\256 SFTP Download Utility.pdf’ did not match any files</p>
<p>&gt;git rm “Resources\DC\DC� SFTP Download Utility.pdf”<br>
fatal: pathspec ‘Resources\DC\DC� SFTP Download Utility.pdf’ did not match any files</p>
<p>&gt;git rm “Resources/DC/DC® SFTP Download Utility.pdf”<br>
fatal: pathspec ‘Resources/DC/DC® SFTP Download Utility.pdf’ did not match any files</p>
<p>&gt;git restore “Resources/DC/DC\256 SFTP Download Utility.pdf”<br>
error: pathspec ‘Resources/DC/DC\256 SFTP Download Utility.pdf’ did not match any file(s) known to git</p>
</blockquote>
<p>Don’t know what to do with this.  Big problem currently cause I’ve got files stashed that I can’t restore because changes are present on the branch.</p>
<p>I’d appreciate any help that might be offered; I’m stumped.</p>
<p>Remi</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there! :wave: Welcome to the Community!</p>
<p>Did you manage to get to the bottom of this?</p>
<p>Looking at your command line output, it’s very strange that you weren’t able to stage the deletion of the file in order to commit it! Did you try using <code>git commit -a</code>, which should stage all changes?</p>
<p>Ah, I see that you wrote in to GitHub Support and were able to resolve this by using a wildcard character, as described here:</p>
<aside class="onebox stackexchange">
  <header class="source">
      <a href="https://stackoverflow.com/questions/23808611/git-checkout-remove-file-with-special-characters" target="_blank">stackoverflow.com</a>
  </header>
  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/3607973/user3607973" target="_blank">
    <img alt="user3607973" src="https://user-images.githubusercontent.com/19803303/181068991-e9b000aa-e33d-4c2e-899e-a3811b991c85.png" class="thumbnail onebox-avatar" width="128" height="128">
  </a>
<h4>
  <a href="https://stackoverflow.com/questions/23808611/git-checkout-remove-file-with-special-characters" target="_blank">Git checkout/remove file with special characters</a>
</h4>

<div class="tags">
  <strong>macos, git, character-encoding</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/3607973/user3607973" target="_blank">
    user3607973
  </a>
  on <a href="https://stackoverflow.com/questions/23808611/git-checkout-remove-file-with-special-characters" target="_blank">01:40PM - 22 May 14 UTC</a>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
